nixos/modules/system/boot/kernel.nix

```diff
@@ -83,7 +83,10 @@ in
     };
 
     boot.kernelParams = mkOption {
-      type = types.listOf types.str;
+      type = types.listOf (types.strMatching ''([^" ]*|"[^"]*")*'' // {
```

Comment:
this should check for all code points `isspace()` detects as spaces. maybe for all spaces in general, better to be a little overzealous in having people add quotes than create bad command lines

Reply:
sounds reasonable, will change to \s when back home